### PR TITLE
integration: check that host in config is a valid host

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Source of confusion otherwise, since invalid hosts won't ever match and UrlReader will fall back to plain fetch, probably resulting in 401 or 404s

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: packages/integration/src/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Checks whether the given url is a valid host */
+export function isValidHost(url: string): boolean {
+  const check = new URL('http://example.com');
+  check.host = url;
```
Comment (freben) [inline packages/integration/src/helpers.ts:20]: does this never-ever throw? :) interesting use of core facilities

**Review by freben — APPROVED** [review]

File: packages/integration/src/helpers.ts (hunk 1 above)
Comment (Rugvip) [reply packages/integration/src/helpers.ts:20]: Nah, just silently ignores any invalid input :>

Comment (freben) [reply packages/integration/src/helpers.ts:20]: i realised that it's a bit harsh to reject punycode things ('fågel.se' becoming 'xn--fgel-qoa.se' which is absolutely fine), hmm

Comment (Rugvip) [reply packages/integration/src/helpers.ts:20]: It does support it, but you need to enter the punycode version in config. That's how the implementation works anyway so should have the same validation in the config

Comment (freben) [reply packages/integration/src/helpers.ts:20]: that makes sense!

**Pull request merged**